### PR TITLE
Add parameter to load specific forms in read_redcap_tidy

### LINKED ISSUE #36: [FEATURE] REDCapTidieR Option to Only Load Specific Forms
# Feature Request Description

For large databases, it is not desirable to extract everything into a `REDCapTidieR` output (the current default behavior) from `REDCapR::redcap_read_oneshot`. Users should have the option to specify forms for extraction to minimize load times and memory use.

# Proposed Solution

There should be an option in `read_redcap_tidy` where users specify forms. This should trickle to `redcap_read_oneshot` for the extraction.

### Checklist

- [x] The issue is atomic
- [x] The issue description is documented
- [x] The issue title describes the problem succinctly
- [x] Developers are assigned to the issue
- [x] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR implements a feature allowing users to specify which forms to download from a REDCap project. If the user specifies any forms that don't exist in the project an informative error message is given.

I also confirmed that this plays well with our check for forms that the user doesn't have access to. If a user requests a form they don't have access to in `forms` they'll get a warning from `check_user_rights()` (same as before). If they specify only forms that they have access to they won't get a warning.

# Proposed Changes 

- Add `forms` parameter to `read_redcap_tidy()` to be passed through to `REDCapR::read_redcap_oneshot()`
- Check argument to `forms` to ensure all forms exist in the project metadata

### Issue Addressed
Closes #36

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Review by skadauke — CHANGES_REQUESTED** [review]
This looks great! Would suggest implementing the error message using CLI.

File: R/checks.R [hunk 1]
```diff
@@ -136,3 +136,37 @@ check_redcap_populated <- function(
...
+#' @param db_metadata The metadata file read by
+#' \code{REDCapR::redcap_metadata_read()}
+#' @param forms The character vector of form names passed to
+#' \code{read_redcap_tidy()}
+#'
+#' @keywords internal
+check_forms_exist <- function(db_metadata, forms) {
+  missing_forms <- setdiff(forms, unique(db_metadata$form_name))
+
+  if (length(missing_forms) > 0) {
```
[the remark this answers is not in the document]
Comment (rsh52) [reply R/checks.R:163]: `cli` looks great, powerful and also aesthetically engaging. 

Should we implement `cli` to just the error message for this PR, to all messages, or separate this feature implementation and then apply `cli` in a separate PR?

**Review by rsh52 — COMMENTED** [review]
I think all of the changes look great, once we decide how/where to implement `cli` should be good to go!

We also have to add @ezraporter to our author list!

**Comment by skadauke** [comment]
> I think all of the changes look great, once we decide how/where to implement `cli` should be good to go!

I would say, let's implement it here, then create an Issue to implement it throughout?

> We also have to add @ezraporter to our author list!

Yes we do!!

**Comment by ezraporter** [comment]
I've updated the error message to use `cli`:

```r
devtools::load_all()

classic_token <- Sys.getenv("REDCAPTIDIER_CLASSIC_API")
redcap_uri <- Sys.getenv("REDCAP_URI")

read_redcap_tidy(redcap_uri,
                 classic_token,
                 forms = "fake-form")
```

cli

Since we're implicitly using `rlang` to issue this error I followed the [Advanced R idea](https://adv-r.hadley.nz/conditions.html#custom-conditions) of adding custom classes to our error:

```r
# Catch the error
e <- tryCatch({
  read_redcap_tidy(redcap_uri,
                   classic_token,
                   forms = "fake-form")
  }, error = function(e) e)

class(e)
#> [1] "form_does_not_exist" "REDCapTidieR_cond" "rlang_error" "error"              
#> [5] "condition"
```

`form_does_not_exist` is specific to this error and the idea would be for `REDCapTidieR_cond` to identify conditions issued by the package in general.

**Comment by skadauke** [comment]
I love the idea of creating custom condition classes using `cli_abort` and then `expect`ing them in the tests!

**Review by skadauke — APPROVED** [review]
Looks great to me!

**Review by rsh52 — APPROVED** [review]
Love it, very succinct and nicely done.
